### PR TITLE
NXP-28939: split the retention based event rule properties

### PR DESCRIPTION
**Pull request by RSalem07**
I open the PR manually, to be able to get the review asap (PO needed asap).

T&P OK: https://qa2.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-saouana-10.10/119/

new T&P after review: https://qa2.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-saouana-10.10/120/

Web-UI PR: https://github.com/nuxeo/nuxeo-web-ui/pull/878/

JIRA: https://jira.nuxeo.com/browse/NXP-28939

### CODE REVIEW

File: addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/RetentionTestCase.java [hunk 1]
```diff
@@ -108,7 +110,7 @@ protected void awaitRetentionExpiration(long millis) throws InterruptedException
 
     protected RetentionRule createRuleWithActions(RetentionRule.ApplicationPolicy policy,
             StartingPointPolicy startingPointPolicy, List<String> docTypes, String startingPointEventId,
-            String startingPointExpression, String matadataXPath, long years, long months, long days,
+            String startingPointExpressionLanguage, String startingPointEventInput, String matadataXPath, long years, long months, long days,
```
Comment (nmpcunha) [inline addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/RetentionTestCase.java:113]: small typo: `matadataXPath`

File: addons/nuxeo-retention/nuxeo-retention-core/src/main/resources/schemas/retention_definition.xsd [hunk 2]
```diff
@@ -31,7 +31,8 @@
       <xs:restriction base="xs:string" ref:resolver="directoryResolver" ref:directory="RetentionEvent" />
     </xs:simpleType>
   </xs:element>
-  <xs:element name="startingPointExpression" type="xs:string" />
+  <xs:element name="startingPointExpressionLanguage" type="xs:string" />
+  <xs:element name="startingPointEventInput" type="xs:string" />
```
Comment (efge) [inline addons/nuxeo-retention/nuxeo-retention-core/src/main/resources/schemas/retention_definition.xsd:35]: This seems unnecessarily verbose to me. I would suggest just `startingPointExpression` and `startingPointValue` instead.

Comment (RSalem07) [reply addons/nuxeo-retention/nuxeo-retention-core/src/main/resources/schemas/retention_definition.xsd]: Yep, it was my first idea and after that I changed it. but let rename them :)

**Comment by RSalem07** [comment]
@nmpcunha thx
> Left just a small note about an already existing typo.

I will do it on master where I can do a separate commit to clean it.

File: addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/RetentionTestCase.java (hunk 1 above)
Comment (RSalem07) [reply addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/RetentionTestCase.java]: thanks, I will do it on master with a separate commit `cleanup/format`

File: addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/RetentionTestCase.java [hunk 3]
```diff
@@ -50,7 +51,7 @@
  * @since 11.1
  */
 @RunWith(FeaturesRunner.class)
-@Features({ TransactionalFeature.class, AutomationFeature.class })
+@Features({ TransactionalFeature.class, AutomationFeature.class, CoreBulkFeature.class })
```
Comment (kevinleturc) [inline addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/RetentionTestCase.java:54]: `CoreBulkFeature` is declared by `CoreFeature`, you should already have it in the test (no needs to depend on it).

File: addons/nuxeo-retention/nuxeo-retention-web-ui/src/main/resources/web/nuxeo.war/ui/document/retentionrule/nuxeo-retentionrule-view-layout.html [hunk 4]
```diff
@@ -69,7 +69,7 @@
         <div role="widget">
           <label>[[i18n("retention.rule.label.startPolicy.eventBased.expression.placeholder")]]</label>
           <nuxeo-tag name="eventInput">
-            [[document.properties.retention_def:startingPointExpression]]
```
Comment (nmpcunha) [inline addons/nuxeo-retention/nuxeo-retention-web-ui/src/main/resources/web/nuxeo.war/ui/document/retentionrule/nuxeo-retentionrule-view-layout.html:72]: Does this pass linting? If yes, it is ok for me!

Comment (RSalem07) [reply addons/nuxeo-retention/nuxeo-retention-web-ui/src/main/resources/web/nuxeo.war/ui/document/retentionrule/nuxeo-retentionrule-view-layout.html:72]: I think yes, I compared it to the others ones, it looks the same formatting :)

File: addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/RetentionTestCase.java [hunk 5]
```diff
@@ -131,23 +134,30 @@ protected RetentionRule createRuleWithActions(RetentionRule.ApplicationPolicy po
     protected RetentionRule createImmediateRuleMillis(RetentionRule.ApplicationPolicy policy, long durationMillis,
             List<String> beginActions, List<String> endActions) {
         return createRuleWithActions(policy, RetentionRule.StartingPointPolicy.IMMEDIATE, Arrays.asList("File"), null,
-                null, null, 0L, 0L, 0L, durationMillis, beginActions, endActions);
+                null, null,null, 0L, 0L, 0L, durationMillis, beginActions, endActions);
```
Comment (efge) [inline addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/RetentionTestCase.java:137]: missing space before `,`

File: addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/TestRetentionManager.java [hunk 6]
```diff
@@ -61,19 +68,19 @@ public void testRuleOnlyFile() {
     @Test
     public void test1DayManualImmediateRuleRunningRetention() throws InterruptedException {
         assertStillUnderRetentionAfter(file, createRuleWithActions(RetentionRule.ApplicationPolicy.MANUAL,
-                RetentionRule.StartingPointPolicy.IMMEDIATE, null, null, null, null, 0, 0, 1, 0, null, null), 1000);
+                RetentionRule.StartingPointPolicy.IMMEDIATE, null, null, null, null,null, 0, 0, 1, 0, null, null), 1000);
```
Comment (efge) [inline addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/TestRetentionManager.java:71]: missing space before `,` (and 2 others below)

File: addons/nuxeo-retention/nuxeo-retention-web-ui/src/main/resources/web/nuxeo.war/ui/document/retentionrule/nuxeo-retentionrule-create-layout.html [hunk 7]
```diff
@@ -116,11 +116,11 @@
              min-chars="0">
            </nuxeo-directory-suggestion>
            <nuxeo-input role="widget" name="eventInput"
-                        label="[[i18n('retention.rule.label.startPolicy.eventBased.expression.description')]]"
-                        placeholder="[[i18n('retention.rule.label.startPolicy.eventBased.expression.placeholder')]]"
-                        name="startingPointExpression"
+                        label="[[i18n('retention.rule.label.startPolicy.eventBased.input.description')]]"
+                        placeholder="[[i18n('retention.rule.label.startPolicy.eventBased.input.placeholder')]]"
```
Comment (efge) [inline addons/nuxeo-retention/nuxeo-retention-web-ui/src/main/resources/web/nuxeo.war/ui/document/retentionrule/nuxeo-retentionrule-create-layout.html:120]: Maybe change the i18n label too? Not sure.

**Pull request merged**